### ISSUE TITLE
[BUG] IDE fails with autocompletion when working with opensearchpy.helpers

### ISSUE THREAD

**Issue by aimran-adroll** [opensearch-project/opensearch-py#426]
### What is the bug?
Trying to debug why a standard editor like intellij/pycharm complains about the opensearch implemented functions, more specifically it thinks functions like `Bool` do not take any arguments!

Nothing major but with many lines of codes, the noise become grating. 

![Screenshot 2023-07-03 at 2 22 20 PM](https://github.com/opensearch-project/opensearch-py/assets/31903245/b575a0b9-dac2-4dd6-86e2-bff7d9a934c1)

### How can one reproduce the bug?
Open a up to date intellij and open up the following snippets

```python
def test_foo():
    from elasticsearch_dsl.query import Bool as EBool
    from opensearchpy.helpers.query import Bool as OBool

    # OpenSearch DSL. Notice the error
    OBool(must=[])

    # Elasticsearch DSL. Notice no errors
    EBool(must=[])
```

### What is the expected behavior?
At the least, intellisense should be able to pick up on docs enough to provide intelligent completions etc.

### What is your host/environment?
```
opensearch-py==2.2.0
elasticsearch-dsl==7.4.1
```
Intellij IDEA 2023.1.3/ OSX 13.4

### Do you have any screenshots?
See above

### Do you have any additional context?
_Add any other context about the problem._

**Comment by dblock** [comment]
@aimran-adroll propose a fix?

**Comment by saimedhi** [comment]
@aimran-adroll, I tried executing above lines of code using pycharm.  I didn't get any error. Can you please try it again once.

**Comment by saimedhi** [comment]
![I notice the same in pycharm](https://github.com/opensearch-project/opensearch-py/assets/117196660/34dce049-6c03-4cdd-9350-cb53bb8300e5)

**Comment by dannosaur** [comment]
The .pyi files are wildly outdated, hence the intellisense errors. Deleting the .pyi files from your venv (while unorthodox) solves the issue. The proper solution requires a lot more work though - to go through every .pyi file and update them to match the API.

**Comment by aimran-adroll** [comment]
@dblock Without knowing too much about opensearchpy internals, I am leaning towards @dannosaur's assessment. There isn't much there in those stub files from what I can tell.

**Comment by saimedhi** [comment]
Contributors, please feel free to address this issue and submit a PR with enhanced pyi files. Multiple PRs are welcome to achieve this. Your contributions are greatly appreciated!

**Comment by tannerjones4075** [comment]
I would like to work on this issue.

**Comment by tannerjones4075** [comment]
Could you officially assign me to this issue?

**Comment by marcosox** [comment]
> The proper solution requires a lot more work though - to go through every .pyi file and update them to match the API.

There are tools to do this automatically:
- https://github.com/python/mypy/blob/master/mypy/stubgen.py
- https://github.com/edreamleo/make-stub-files

**Comment by saimedhi** [comment]
Hello @marcosox, I've used mypy/stubgen for writing pyi files for the existing opensearch-py helpers in this repo, but we're still facing this issue.

**Comment by tannerjones4075** [comment]
What issues are you facing?

**Comment by saimedhi** [comment]
Hello @tannerjones4075, I wanted to mention that for #287 , which introduced most of the helpers to opensearch-py, I used mypy/stubgen to create the pyi files.

**Comment by dblock** [comment]
I believe this should be resolved in #536 and #564, closing. If you see another type problem, open a new issue.
